Source/ChocolateyGui/App.xaml.cs

```diff
@@ -21,7 +22,20 @@ public partial class App
     {
         static App()
         {
-            Container = AutoFacConfiguration.RegisterAutoFac();
+            Container = AutoFacConfiguration.InitialRegistration();
+
+            var configurationProvider = Container.Resolve<IChocolateyConfigurationProvider>();

```

Comment:
I feel like this should be inside the AutoFac registration logic, rather than having it in App(). No Initial/Remaining, just one intelligent registration. Especially since it doesn't make much sense to do a Resolve when registration knows exactly what to expect.